BUILD

```diff
@@ -0,0 +1,33 @@
+load("//:sdk_config.bzl", "sdk_config_transition")
+load("@bazel_skylib//rules:common_settings.bzl", "string_flag")
+
+cc_library(
+    name = "default_config",
+    hdrs = ["sdk_config.h"],
+    visibility = ["//visibility:public"],
+)
+
+label_flag(
+    name = "sdk_config",
+    build_setting_default = ":default_config",
+    visibility = ["//visibility:public"],
+)
+
+string_flag(
+    name = "sdk_config_dir",
+    build_setting_default = "",
+)
+
+cc_binary(
+    name = "bin",
+    srcs = ["bin.cc"],
+    deps = [":lib"],
+)
+
+cc_library(
+    name = "lib",
+    hdrs = ["lib.h"],
+    includes = ["a"], # Why does replacing "a" with ":sdk_config_dir not work?
```

Comment:
`string_flag` lets you access the result as a `BuildSettingInfo`. `includes` is looking for a list of strings, not a list of labels to extract strings from via their providers.

I would write a little custom rule (or genrule) which copies the file pointed to by `:sdk_config` into a file in a known place (like `sdk_config_dir/sdk_config.h`), and then you can pass `includes = ["sdk_config_dir"].

Some rules (not the C++ ones) have versions of their attrs which take a label, where they use the value of that file instead of a direct string ([pkg_deb](https://github.com/bazelbuild/rules_pkg/tree/master/pkg#pkg_deb).version_file for example). That allows solving this problem with a trivial custom rule/genrule that dumps the string into a file. Unfortunately, that doesn't help here.

Reply:
Would that cause problems if I build 2 binaries with conflicting sdk_config.h files?
bazel build //a:bin //b:bin

Reply:
Yea, you'd have to make a unique path for each one. You need a custom rule to do the transition anyways, so you could put it there, or a separate little rule. `ctx.label.name` is unique by definition, and `ctx.actions.declare_file` only lets you do stuff in the current package, so something like `ctx.actions.declare_file('%s_include_dir/sdk_config.h' % ctx.label.name)` works well.